### DISCUSSION TITLE
Retrieving the Trusted players

### DISCUSSION THREAD

**Discussion by pixar02** [GriefPrevention/GriefPrevention#648] [Q&A] [answered]
I have been asked to add colors to the placeholder:
%griefprevention_currentclaim_ownername% (When a player is trusted it becomes green, if not red)
I was wondering what the best way would be to retrieve that data form the Current claim the player is standing in. The only possibility that I see is to check, is if the AllowBuild(....) equals null....

For clarification I made the PlaceholderAPI expansion for GriefPrevention., which add variables so other plugins can use data form GriefPrevention, without directly conecting to the GriefPrevention instance.

**Comment by RoboMWM** [comment]
You should be able to get the trust map, otherwise there is a method to check each trust group in Claim.

**Comment by pixar02** [comment]
I don't see the method you are talking about, unless I'm totally blind.
The only maps I see (in Claims) are: managers (List not a map), playerIDToClaimPermissionMap  (Map) and children (List of Claims).

**Comment by Jikoo** [comment, marked as answer]
For a specific player:
`Claim#allowAccess(Player) == null` for access trust, `Claim#allowContainers(Player) == null` for container trust, and `Claim#allowBuild(Player, Material) == null` for full trust. The material being built with should be a non-crop block.

~~If you don't want your color to change when the player is ineligible for certain levels of trust (in PVP, under siege, etc.) you can check `Claim#hasExplicitPermission(Player, ClaimPermission)` instead. Be aware that you will also need to check parent claims' permissions unless `Claim#getSubclaimRestrictions` is true.~~ Oops, method is private.

For retrieving all players with access:
`Claim#getPermissions(ArrayList<String>, ArrayList<String>, ArrayList<String>, ArrayList<String>)`
The provided lists will be populated with the trusted users for the corresponding levels.

The trust map Robo mentioned is the one you found, however, it has private access.

**Comment by pixar02** [comment]
I take it that the normal command '/trust <player>' gives the build level.
ant takes this order:
- '/AccessTrust' : BUILD (Like the '/trust')
- '/ContainerTrust' : INVENTORY
- '/PermissionTrust' : ACCESS

**Comment by bigpresh** [comment]
/AccessTrust lets you open doors etc, but does not allow you to open inventory-containing items (chests, etc), nor does it let you place/break blocks.

/ContainerTrust gives you AccessTrust, and also the ability to open/interact with inventory-type blocks, but still no build/break ability (but, IIRC, the ability to place minecarts).

Build Trust (/trust) gives all the above, plus the ability to place/destroy blocks - in other words, full trust.

/PermissionTrust gives you the ability to assign permissions to others, i.e. to manage the trust list of the claim.

The word "trust" now looks weird after typing it so many times :)

**Comment by pixar02** [comment]
So using the enum in the source code, which one is the first one in line.
-Access, inventory, build?
-Build, inventory, access?
-Build, access, inventory?

**Comment by bigpresh** [comment]
If you mean the list returned by Claim#getPermissions, then a quick look at the code reveals:

`public void getPermissions(ArrayList<String> builders, ArrayList<String> containers, ArrayList<String> accessors, ArrayList<String> managers)`

So, build trust, container trust, access trust, permissions trust.

If that's not what you meant, have a look at the code to see what it does, it's all here :)

**Comment by Jikoo** [comment]
`Access` is `/accesstrust`, `Build` is full `/trust`, and `Inventory` is `/containertrust`.
Access < Container < Build in terms of ability to modify the claim's content.

Permission trust is entirely separate and does not grant any kind of access to the blocks in the claim, instead allowing the user to manage the trust of other users up to their own trust level. Think of it like an additional group not in the default track in a permissions plugin.

**Comment by pixar02** [comment]
Ok, thank you all!
